.github/workflows/build.yaml

```diff
@@ -20,6 +22,7 @@ jobs:
     steps:
       - uses: actions/checkout@v4
         with:
+          ref: ${{ github.event.pull_request.head.sha || github.ref }}
```

Comment:
What's the purpose of this line?

Reply:
Its related to the previous comment - if you want to run on the pull request head commit, you have to use the `github.event.pull_request.head.sha` otherwise it will run on the main branch from this repository - https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request